__tests__/auth.test.js

```diff
@@ -0,0 +1,56 @@
+const pool = require('../lib/utils/pool');
+const setup = require('../data/setup');
+const request = require('supertest');
+const app = require('../lib/app');
+const GithubUser = require('../lib/models/GithubUser');
+
+jest.mock('../lib/utils/github');
+
+describe('backend-gitty routes', () => {
+  beforeEach(() => {
+    return setup(pool);
+  });
+
+  afterAll(() => {
+    pool.end();
+  });
+
+  it('redirect to the github oauth when logged in', async () => {
+    const req = await request(app).get('/api/v1/github/login');
+    expect(req.header.location).toMatch(
+      /https:\/\/github.com\/login\/oauth\/authorize\?client_id=[\w\d]+&scope=user&redirect_uri=http:\/\/localhost:7890\/api\/v1\/github\/login\/callback/i
+    );
+  });
+
+  it('should login and redirect users to /api/v1/github/dashboard', async () => {
+    const req = await request
+      .agent(app)
+      .get('/api/v1/github/login/callback?code=42')
+      .redirects(1);
+
+    expect(req.body).toEqual({
+      id: expect.any(String),
+      username: 'fake_github_user',
+      email: 'not-real@example.com',
+      iat: expect.any(Number),
+      exp: expect.any(Number),
+    });
+  });
+
+  it('should sign out a user', async () => {
+    const agent = request.agent(app);
+    await GithubUser.insert({
```

Comment:
We shouldn't create users like this. We should hit the /login/callback controller route which will take care of that for us.